crates/swc_ecma_parser/src/lexer/state.rs

```diff
@@ -22,6 +22,7 @@ use crate::{
 #[derive(Clone)]
 pub(super) struct State {
     pub is_expr_allowed: bool,
```

Comment:
Not sure why we need it, because I looked at babel and esbuild logic and don't found the same semantic, so I think it can be simplified